business/transaction_service/cancel.go

```diff
@@ -2,11 +2,45 @@ package transaction_service
 
 import (
 	"context"
+	"errors"
+	"fmt"
 
 	"mock-payment-provider/business"
+	"mock-payment-provider/primitive"
+	"mock-payment-provider/repository/transaction"
 )
 
 func (d Dependency) Cancel(ctx context.Context, orderId string) (business.CancelResponse, error) {
```

Comment:
why not pointer? can you explain or link to read?

because I think in line 19 and 34 it will use new object instead of using existing object (in this case Dependency)

Reply:
diluar scope PR ini. tapi iya emang harusnya pointer.